### DISCUSSION TITLE
Cuda memory error

### DISCUSSION THREAD

**Discussion by oscarbau** [opengeos/segment-geospatial#149] [Q&A] [answered]
Hi all,

I am testing the use of Geosam with Cuda enabled.

When I try to run the segmentation I get the following error:
![image](https://github.com/opengeos/segment-geospatial/assets/41731874/549c9d12-cd28-41c6-9fc3-33513eb945eb)

CUDA out of memory. Tried to allocate 3.52 GiB (GPU 0; 10.00 GiB total capacity; 6.87 GiB already allocated; 0 bytes free; 6.98 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

Any ideas on how to handle this?

Thanks!

**Comment by giswqs** [comment, marked as answer]
Try a smaller image or run `sam.clear_cuda_cache()`
